modules/queue/base_redis_test.go

```diff
@@ -56,7 +56,7 @@ func TestBaseRedis(t *testing.T) {
 	}()
 	if !waitRedisReady("redis://127.0.0.1:6379/0", 0) {
 		redisServer = redisServerCmd(t)
-		if redisServer == nil && os.Getenv("CI") != "" {
+		if redisServer == nil || os.Getenv("CI") != "" {
```

Comment:
I would suggest not doing so.

Keep the old logic will remind developers that "please remember to test the redis queue"

Since in most cases, developers only run single test, this test doesn't affect them.

And there is always a chance for developers to bypass it: `CI=local make test-xxxx`

Reply:
Maybe I'm mis-understanding the use of the `CI` env variable. The problem with the existing logic on main is that it will skip the test with `CI=1` (or anything else).

The problem with the logic suggested here is that it will just always skip the test :)

Please see my suggestion below. I think the default should be local and we set the CI env flag in CI.

Reply:
It is not reasonable to force developers to install redis in order to run tests.